### ISSUE TITLE
Issue with knife tag create 

### ISSUE THREAD

**Issue by manjukdvg** [chef/chef#3534]
I tried to create a tag for a node . Getting the following error . 

```
knife tag create prod-server-4 prodapi -VV
INFO: Using configuration from /Users/manju/.chef/knife.rb
DEBUG: Chef::HTTP calling Chef::HTTP::JSONInput#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::JSONToModelOutput#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::CookieManager#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::Decompressor#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::Authenticator#handle_request
DEBUG: Signing the request as manju
DEBUG: Chef::HTTP calling Chef::HTTP::RemoteRequestID#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::ValidateContentLength#handle_request
DEBUG: Initiating GET to https://<CHEF_SERVER_URL>/nodes/prod-server-4
DEBUG: ---- HTTP Request Header Data: ----
DEBUG: Accept: application/json
DEBUG: Accept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3
DEBUG: X-OPS-SIGN: algorithm=sha1;version=1.0;
DEBUG: X-OPS-USERID: manju
DEBUG: X-OPS-TIMESTAMP: 2015-06-15T06:17:16Z
DEBUG: X-OPS-CONTENT-HASH: 2jmj7l5rSw0yVb/vlWAYkK/YBwk=
DEBUG: X-OPS-AUTHORIZATION-1: w07UQXeVSmYg0vCllrGgt4tqd+Q+yJjIoc2UR+rC+8NxMRF50xR/S1r
DEBUG: X-OPS-AUTHORIZATION-2: pA0Abp+7oA37KcmDUxAa10AcmQbNkdCuQSI3aLP9a/WVTayyJwGOao8
DEBUG: X-OPS-AUTHORIZATION-3: SrJhHpxxyj/JVx6prd6PQlZDsc1PKfZT9XoY0wXL17r21y27cP+uM6v
DEBUG: X-OPS-AUTHORIZATION-4: dD/jp8p7XIpnSozTUcdFVI/mwWP5gdosBBCEfDXAXniwGZZj7Br5/3Z
DEBUG: X-OPS-AUTHORIZATION-5: mCJN6WTT/uJ+lBhw7nwS+2+1Hv+MAm6svcw4kwdspQzsLfSLuaJ4l3j
DEBUG: X-OPS-AUTHORIZATION-6: thp6yqUOd9dwwqc8lduywOmOiuSzO0qN/R+Eb4Ed31Q==
DEBUG: HOST: <cCHEF_SERVER_URL>:443
DEBUG: X-REMOTE-REQUEST-ID: 38cd9a55-6e9c-434t-a8fa-a660f0ab6f83
DEBUG: ---- End HTTP Request Header Data ----
DEBUG: ---- HTTP Status and Header Data: ----
DEBUG: HTTP 1.1 200 OK
DEBUG: server: nginx/1.4.4
DEBUG: date: Mon, 15 Jun 2015 06:16:54 GMT
DEBUG: content-type: application/json
DEBUG: transfer-encoding: chunked
DEBUG: connection: close
DEBUG: x-ops-api-info: flavor=osc;version=11.0.2;erchef=1.4.1
DEBUG: content-encoding: gzip
DEBUG: ---- End HTTP Status/Header Data ----
DEBUG: Chef::HTTP calling Chef::HTTP::ValidateContentLength#handle_response
DEBUG: HTTP server did not include a Content-Length header in response, cannot identify truncated downloads.
DEBUG: Chef::HTTP calling Chef::HTTP::RemoteRequestID#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::Authenticator#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::Decompressor#handle_response
DEBUG: decompressing gzip response
DEBUG: Chef::HTTP calling Chef::HTTP::CookieManager#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::JSONToModelOutput#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::JSONInput#handle_response
/opt/chefdk/embedded/apps/chef/lib/chef/knife/tag_create.rb:45:in `block in run': undefined method `<<' for nil:NilClass (NoMethodError)
    from /opt/chefdk/embedded/apps/chef/lib/chef/knife/tag_create.rb:44:in `each'
    from /opt/chefdk/embedded/apps/chef/lib/chef/knife/tag_create.rb:44:in `run'
    from /opt/chefdk/embedded/apps/chef/lib/chef/knife.rb:417:in `block in run_with_pretty_exceptions'
    from /opt/chefdk/embedded/apps/chef/lib/chef/local_mode.rb:38:in `with_server_connectivity'
    from /opt/chefdk/embedded/apps/chef/lib/chef/knife.rb:416:in `run_with_pretty_exceptions'
    from /opt/chefdk/embedded/apps/chef/lib/chef/knife.rb:213:in `run'
    from /opt/chefdk/embedded/apps/chef/lib/chef/application/knife.rb:139:in `run'
    from /opt/chefdk/embedded/apps/chef/bin/knife:25:in `<top (required)>'
    from /usr/bin/knife:40:in `load'
    from /usr/bin/knife:40:in `<main>'
```

**Comment by lamont-granquist** [comment]
My guess is that this node hasn't successfully converged or otherwise doesn't have a tags attribute at all and the knife code is missing the initialization that normally happens in the chef-client run to set this up as an empty array.  Then the NoMethod `<<` on NilClass occurs.

**Comment by manjukdvg** [comment]
Yes , It was not tagged while converging . Adding tag solved the issue . Thanks.

**Comment by josb** [comment]
I'm running into this too, with chef-client 12.5.1. Possible fix:
```
--- a/lib/chef/node.rb
+++ b/lib/chef/node.rb
@@ -378,10 +378,10 @@ class Chef

     def tag(*tags)
       tags.each do |tag|
-        self.normal[:tags].push(tag.to_s) unless self[:tags].include? tag.to_s
+        tags.push(tag.to_s) unless tags.include? tag.to_s
       end

-      self[:tags]
+      tags
     end

     # Extracts the run list from +attrs+ and applies it. Returns the remaining attributes
```

**Comment by thommay** [comment]
This was fixed by #4064
